Ewald.js

```diff
@@ -0,0 +1,120 @@
+    var bankSelect = document.getElementById('bankSelect');
+    var gekozenBank = bankSelect.options[bankSelect.selectedIndex].value;
+
+    var bankCode           = bepaalBankCode(gekozenBank);
+    rekeningNummer         = bepaalHeleRekeningNummer(rekeningNummer);
+
+    var controleGetal      = bepaalControleGetal(bankCode, landCode, rekeningNummer);
+
+    document.getElementById("iban").innerHTML = 'Je IBAN is: ' + landCode + controleGetal + bankCode + rekeningNummer;
+}
+
+// de bankCode kun je halen uit de globale 'bankIdentificatie' array.
+// We zouden de bankCode natuurlijk ook gewoon in de <option>s van de <select> kunnen plaatsen..
+// maar dan hoef je niet te oefenen met arrays :)
+function bepaalBankCode(bank) {
+    var bankCode = '';
+    for(var i=0; i < bankIdentificatie.length; i++){
+    	if(bankIdentificatie[i][0]===bank){
+				bankCode = bankIdentificatie[i][1];
+      }
+    }
+    // haal hier de bankCode op uit 'bankIdentificatie'
+
+    return bankCode;
+}
+
+// Maak van het 7- of 9-cijferige nummer een 10-cijferig rekeningnummer
+function bepaalHeleRekeningNummer(rekeningNummer) {
+
+    // zorg er hier voor dat het rekeningnummer altijd 10 cijfers lang is, eventueel met nullen ervoor
+    while(rekeningNummer.length < 10){
+      rekeningNummer = '0'+rekeningNummer;
+    }
+
+    while(rekeningNummer.length > 10){
+      rekeningNummer = rekeningNummer.slice(0, 10);
+    }
+
+    return rekeningNummer;
+}
+
+// bereken nu het controlegetal
+//    Neem de rekeningidentificatie, plaats de landcode daarachter
+//    Vervang alle letters door twee cijfers gebaseerd op de volgorde in het alfabet (A=10, B=11, C=12, ..., Z=35)
+//    Voeg twee nullen toe aan het einde
+//    Deel dit door 97 en neem de rest (mod97)
+//    Het controlegetal is nu 98 - <deze rest>
+//    Is het controlegetal kleiner dan 10? Zet er dan een 0 voor. (2 wordt dus 02)
+//    Zie http://nl.wikipedia.org/wiki/International_Bank_Account_Number#Controlegetal
+function bepaalControleGetal(bankCode, landCode, rekeningNummer) {
+    var controleGetal = 0;
+    var nummerAlsString = bankCode + rekeningNummer + landCode;
+    // In de geconverteerdeString zijn letters vervangen door hun getalwaarde
+    // Hierbij moet A de waarde 10 krijgen, B wordt 11, C = 12, etc)
+    var geconverteerdeString = '';
+
+    // Een simpele manier om dit te doen is b.v. door elk karakter in nummerAlsString te bekijken en
+    // te controleren of dit een getal is of niet (met isNaN: isNaN(3) = false, isNaN('d') = true).
+    // Vervolgens kun je getallen direct toevoegen aan de geconverteerde string, letters moet je eerst omzetten
+    // voordat je ze toe kunt voegen.
+    // Hint: 'ABCDEFG'.charCodeAt(2) - 'A'.charCodeAt(0) geeft als resultaat 2, want 'C' - 'A' = 67 - 65 = 2;
+    // A moet 10 worden, B = 11, C = 12, etc.
+
+    for(var i=0;i<nummerAlsString.length;i++){
+      var letter = nummerAlsString.charAt(i);
+      if(isNaN(nummerAlsString.charAt(i))){

```

Comment:
Dit gaat nog niet goed! isNaN(nummerAlsString.charAt(i)) is altijd true. (waarom?)